diff --git a/node_modules/nativewind/dist/styled/use-interaction.js b/node_modules/nativewind/dist/styled/use-interaction.js
index a66e32b..bc46428 100644
--- a/node_modules/nativewind/dist/styled/use-interaction.js
+++ b/node_modules/nativewind/dist/styled/use-interaction.js
@@ -25,6 +25,25 @@ function useInteraction(dispatch, mask, props) {
                     dispatch({ type: "active", value: false });
                 };
             }
+            else if (react_native_1.Platform.isTV === true) {
+                handlers.onPress = function (event) {
+                    if (ref.current.onPress) {
+                        ref.current.onPress(event);
+                    }
+                    dispatch({ type: "active", value: true });
+                    setTimeout(() => dispatch({ type: "active", value: false }), 200);
+                };
+                handlers.onLongPress = function (event) {
+                    if (ref.current.onLongPress) {
+                        ref.current.onLongPress(event);
+                    }
+                    if (event.eventKeyAction === 0) {
+                      dispatch({ type: "active", value: true });
+                    } else {
+                      dispatch({ type: "active", value: false });
+                    }
+                };
+            }
             else {
                 handlers.onPressIn = function (event) {
                     if (ref.current.onPressIn) {
