diff --git a/node_modules/react-native/Libraries/Components/Pressable/Pressable.d.ts b/node_modules/react-native/Libraries/Components/Pressable/Pressable.d.ts
index cb0797f..597cc50 100644
--- a/node_modules/react-native/Libraries/Components/Pressable/Pressable.d.ts
+++ b/node_modules/react-native/Libraries/Components/Pressable/Pressable.d.ts
@@ -20,6 +20,7 @@ import {
 import {View} from '../View/View';
 import {AccessibilityProps} from '../View/ViewAccessibility';
 import {ViewProps} from '../View/ViewPropTypes';
+import type {TVParallaxPropertiesType} from '../TV/TVViewPropTypes';
 
 export interface PressableStateCallbackType {
   readonly pressed: boolean;
@@ -159,6 +160,13 @@ export interface PressableProps
    * Duration (in milliseconds) to wait after press down before calling onPressIn.
    */
   unstable_pressDelay?: number | undefined;
+
+  /**
+   * *(Apple TV only)* Object with properties to control Apple TV parallax effects.
+   *
+   * @platform ios
+   */
+  tvParallaxProperties?: TVParallaxPropertiesType,
 }
 
 // TODO use React.AbstractComponent when available
diff --git a/node_modules/react-native/Libraries/Components/Pressable/useAndroidRippleForView.js b/node_modules/react-native/Libraries/Components/Pressable/useAndroidRippleForView.js
index 20dd19e..009ba49 100644
--- a/node_modules/react-native/Libraries/Components/Pressable/useAndroidRippleForView.js
+++ b/node_modules/react-native/Libraries/Components/Pressable/useAndroidRippleForView.js
@@ -80,8 +80,8 @@ export default function useAndroidRippleForView(
           if (view != null) {
             Commands.hotspotUpdate(
               view,
-              event.nativeEvent.locationX ?? 0,
-              event.nativeEvent.locationY ?? 0,
+              event.nativeEvent?.locationX ?? 0,
+              event.nativeEvent?.locationY ?? 0,
             );
             Commands.setPressed(view, true);
           }
@@ -91,8 +91,8 @@ export default function useAndroidRippleForView(
           if (view != null) {
             Commands.hotspotUpdate(
               view,
-              event.nativeEvent.locationX ?? 0,
-              event.nativeEvent.locationY ?? 0,
+              event.nativeEvent?.locationX ?? 0,
+              event.nativeEvent?.locationY ?? 0,
             );
           }
         },
