diff --git a/node_modules/react-native-bottom-tabs/ios/PageSelectedEvent.swift b/node_modules/react-native-bottom-tabs/ios/PageSelectedEvent.swift
index 471218e..fa69058 100644
--- a/node_modules/react-native-bottom-tabs/ios/PageSelectedEvent.swift
+++ b/node_modules/react-native-bottom-tabs/ios/PageSelectedEvent.swift
@@ -27,7 +27,7 @@ import React
   @objc public func arguments() -> [Any] {
     return [
       viewTag,
-      RCTNormalizeInputEventName(eventName),
+      RCTNormalizeInputEventName(eventName) ?? eventName,
       [
         "key": key
       ]
diff --git a/node_modules/react-native-bottom-tabs/ios/TabLongPressedEvent.swift b/node_modules/react-native-bottom-tabs/ios/TabLongPressedEvent.swift
index ddb2337..493f4de 100644
--- a/node_modules/react-native-bottom-tabs/ios/TabLongPressedEvent.swift
+++ b/node_modules/react-native-bottom-tabs/ios/TabLongPressedEvent.swift
@@ -31,7 +31,7 @@ protocol RCTEvent {}
   @objc public func arguments() -> [Any] {
     return [
       viewTag,
-      RCTNormalizeInputEventName(eventName),
+      RCTNormalizeInputEventName(eventName) ?? eventName,
       [
         "key": key
       ]
diff --git a/node_modules/react-native-bottom-tabs/ios/TabViewImpl.swift b/node_modules/react-native-bottom-tabs/ios/TabViewImpl.swift
index 8295d99..e201367 100644
--- a/node_modules/react-native-bottom-tabs/ios/TabViewImpl.swift
+++ b/node_modules/react-native-bottom-tabs/ios/TabViewImpl.swift
@@ -59,6 +59,7 @@ struct TabViewImpl: View {
         renderTabItem(at: index)
       }
     }
+    #if !os(tvOS)
     .onTabItemEvent({ index, isLongPress in
       guard let key = props.items.filter({
         !$0.hidden || $0.key == props.selectedPage
@@ -72,6 +73,7 @@ struct TabViewImpl: View {
         emitHapticFeedback()
       }
     })
+    #endif
     .tintColor(props.selectedActiveTintColor)
     .getSidebarAdaptable(enabled: props.sidebarAdaptable ?? false)
     .configureAppearance(props: props)
@@ -82,6 +84,9 @@ struct TabViewImpl: View {
           UIView.setAnimationsEnabled(true)
         }
       }
+      #if os(tvOS)
+      onSelect(newValue)
+      #endif
     }
   }
 
@@ -136,7 +141,7 @@ struct TabViewImpl: View {
   }
 }
 
-@available(iOS 15.0, *)
+@available(iOS 15.0, tvOS 15.0, *)
 private func configureAppearance(for appearanceType: String, appearance: UITabBarAppearance) -> UITabBarAppearance {
   if (appearanceType == "transparent") {
     return appearance
@@ -178,7 +183,7 @@ private func updateTabBarAppearance(props: TabViewProps) {
   )
 
 
-  if #available(iOS 15.0, *) {
+  if #available(iOS 15.0, tvOS 15.0, *) {
     appearance = configureAppearance(for: props.scrollEdgeAppearance ?? "", appearance: appearance)
 
     if props.translucent == false {
@@ -294,7 +299,7 @@ extension View {
   func tintColor(_ color: UIColor?) -> some View {
     if let color {
       let color = Color(color)
-      if #available(iOS 16.0, *) {
+      if #available(iOS 16.0, tvOS 16.0, *) {
         self.tint(color)
       } else {
         self.accentColor(color)
@@ -308,7 +313,7 @@ extension View {
   // By default they are always filled.
   @ViewBuilder
   func noneSymbolVariant() -> some View {
-    if #available(iOS 15.0, *) {
+    if #available(iOS 15.0, tvOS 15.0, *) {
       self
         .environment(\.symbolVariants, .none)
     } else {
